src/net/tcp/stream.rs

```diff
@@ -113,11 +113,19 @@ impl TcpStream {
             },
         )
     }
+
+    pub async fn peek(&self, buf: &mut [u8]) -> io::Result<usize> {
+        let n = World::current(|world| {
+            let host = world.current_host_mut();
+            host.tcp.peek_buffer(self.read_half.pair.as_ref(), buf)
+        });
+        Ok(n)
+    }
```

Comment:
This is not correct, data needn't already be there. A correct implementation would wait for data to be there, and then return the data without advancing the internal cursor

Reply:
Thanks for the feedback, I missed that behavior. I'll close this in favor of your draft PR, #64.